src/hotspot/cpu/x86/x86.ad

```diff
@@ -1881,7 +1882,11 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
       }
       break;
     case Op_VectorCastF2X:
-      if (is_subword_type(bt) || bt == T_LONG) {
+      if (((UseAVX <= 2) || (!VM_Version::supports_avx512vl())) &&
+          (is_subword_type(bt) || bt == T_LONG)) {
+        return false;
+      }
+      if ((bt == T_LONG) && !VM_Version::supports_avx512dq()) {
```

Comment:
Again overlapping conditions. So T_LONG requires both: AVX512, avx512vl and avx512dq?

What about T_INT?

Reply:
T_INT doesn't need AVX512dq. Float to long conversion (T_LONG) uses evcvttps2qq, which needs AVX512dq.

Reply:
Okay. I see that there are 2 instructions to support F2I by using avx or evex encoding. They cover all cases.
No you are introducing sub_integer and long types only for evex encoding.

You need comment that F2I is supported in all cases. For other integral types you need 512vl  and additionally 512dq for T_LONG.

Note, you don't need to check (UseAVX <= 2) because avx512vl bit is cleaned in such case. It is the same for VectorCastD2X code.

In such case I suggest:

```
  if (is_subword_type(bt) && !VM_Version::supports_avx512vl() ||
      (bt == T_LONG) && !VM_Version::supports_avx512vldq()) {
    return false;
  }
```